src/pages/Facility/settings/devices/components/ServiceHistoryForm.tsx

```diff
@@ -0,0 +1,197 @@
+      },
+    }),
+    onSuccess: (resp: ServiceHistory) => {
+      toast.success(t("service_record_updated_successfully"));
+      form.reset();
+      onSubmitSuccess(resp);
+    },
+  });
+
+  const isPending = isCreatePending || isUpdatePending;
+
+  const onSubmit = (values: FormValues) => {
+    const payload = {
+      ...values,
+      serviced_on: values.serviced_on.toISOString(),
+    };
+
+    if (isEditMode) {
+      updateService(payload);
+    } else {
+      createService(payload);
+    }
+  };
+
+  return (
+    <Form {...form}>
+      <form onSubmit={form.handleSubmit(onSubmit)} className="space-y-4">
+        <FormField
+          control={form.control}
+          name="serviced_on"
+          render={({ field }) => (
+            <FormItem className="flex flex-col">
+              <FormLabel required>{t("service_date")}</FormLabel>
+              <Popover>
+                <PopoverTrigger asChild>
+                  <FormControl>
+                    <Button
+                      variant={"outline"}
+                      className={cn(
+                        "w-full pl-3 text-left font-normal",
+                        !field.value && "text-gray-500",
+                      )}
+                      data-cy="service-date-select"
+                    >
+                      {field.value ? (
+                        format(field.value, "PPP")
+                      ) : (
+                        <span>{t("select_date")}</span>
+                      )}
+                      <CalendarIcon className="ml-auto h-4 w-4 opacity-50" />
+                    </Button>
+                  </FormControl>
+                </PopoverTrigger>
+                <PopoverContent className="w-auto p-0" align="start">
+                  <Calendar
+                    mode="single"
+                    selected={field.value}
+                    onSelect={field.onChange}
+                  />
+                </PopoverContent>
+              </Popover>
+              <FormMessage />
+            </FormItem>
+          )}
+        />
+        <FormField
+          control={form.control}
+          name="note"
+          render={({ field }) => (
+            <FormItem>
+              <FormLabel required>{t("service_notes")}</FormLabel>
+              <FormControl>
+                <Textarea
+                  data-cy="service-notes-input"
+                  placeholder={t("service_notes_enter")}
+                  {...field}
+                  rows={5}
+                />
+              </FormControl>
+              <FormMessage />
+            </FormItem>
+          )}
+        />
+        <div className="flex justify-end space-x-2 pt-4">
+          <Button type="submit" disabled={isPending} data-cy="submit-button">
+            {isPending
+              ? isEditMode
+                ? t("updating")
+                : t("saving")
+              : isEditMode
+                ? t("update")
+                : t("save")}
```

Comment:
save is same as update or create.

it should either be both "save" when editing or creating. or specifically keep "create" or "update".

Reply:
This for showing the pending state (so saving/updating when it's calling query or otherwise).